app/api/hooks/useApi.js

```diff
@@ -0,0 +1,19 @@
+import { useState } from "react";
+
+export default useApi = (apiFunc) => {
+  const [data, setData] = useState([]);
+  const [error, setError] = useState(false);
+  const [loading, setLoading] = useState(false);
```

Comment:
These 3 state can be combined to one say `status` and we can reuse the AsyncStatus logic here. which I described above.

Reply:
Updated